programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/adapter/VFSFileObjectAdapter.java

```diff
@@ -601,29 +610,19 @@ private static boolean checkIfFileWritable(DataSpacesFileObject dsfo) throws Fil
         if (!dsfo.isWritable()) {
             return false;
         }
-        File file = null;
+
         try {
-            file = convertDataSpaceToFileIfPossible(dsfo);
+            File file = convertDataSpaceToFileIfPossible(dsfo);
+
+            if (file != null) {
+                return Files.isWritable(file.toPath());
+            }
         } catch (URISyntaxException e) {
             throw new FileSystemException(e);
         } catch (DataSpacesException e) {
             throw new FileSystemException(e);
         }
-        if (file != null) {
-            boolean answer = file.canWrite();
-            if (!answer) {
-                return false;
-            }
-            // in some cases the canWrite method is not accurate, we try to create a file to check real write permission
-            File tryFile = new File(file, "testWrite.ack");
```

Comment:
Here is the java bug report describing the issue, with the resolution "Won't fix". 

http://bugs.java.com/bugdatabase/view_bug.do?bug_id=4420020

Basically, java only checks the read-only attribute of the file, which is completely wrong and does not take care of OS ACL security.

One stack overflow user suggested to use java.nio.file.Files#isWritable but I haven't given it a try.

The new java paths API include some ACL-related methods which could eventually be used, with the drawback of additional code complexity.